### ISSUE TITLE
Abigen with -pkg option cannot parse contract JSON from stdin

### ISSUE THREAD

**Issue by vsbogd** [ethereum/go-ethereum#19292]
After PR https://github.com/ethereum/go-ethereum/pull/17648 it is not possible to use ```--pkg``` flag with JSON contract artifacts. 

Previously execution of command 
```
cat <contract.json> | abigen --abi -  --pkg <packagename> --out <output.go>
```
lead to execution of the ```contractsFromStdin()``` at https://github.com/ethereum/go-ethereum/blob/7e7781ffaa5c5486c5e7b66dea4986e063524f63/cmd/abigen/main.go#L94

And after the PR ```pkgFlag == ""``` condition prevents falling into this branch:
https://github.com/ethereum/go-ethereum/blob/7e7781ffaa5c5486c5e7b66dea4986e063524f63/cmd/abigen/main.go#L78
And fails with the parse error.

See https://github.com/ethereum/go-ethereum/issues/17647#issuecomment-443110907 and https://github.com/ethereum/go-ethereum/issues/17647#issuecomment-464965285 for discussion

#### System information

Go-ethereum version: 1.8.23
OS & Version: Linux

#### Expected behaviour

Golang stub for the contract is generated.

#### Actual behaviour

Error printed:
Failed to generate ABI binding: json: cannot unmarshal object into Go value of type []struct { Type string; Name string; Constant bool; Anonymous bool; Inputs []abi.Argument; Outputs []abi.Argument }

#### Steps to reproduce the behaviour

```
echo '{"contracts":{"test.sol:Test":{"abi":"[]","bin":"6080604052348015600f57600080fd5b50603580601d6000396000f3fe6080604052600080fdfea165627a7a723058204607ea1e116ed157fa353615ead6dab3b941f6c06adf7c0e724273ad8ca0a9c80029","devdoc":"{\"methods\":{}}","userdoc":"{\"methods\":{}}"}},"version":"0.5.3+commit.10d17f24.Linux.g++"}' | abigen --abi - --pkg test
```

**Comment by vsbogd** [comment]
Raised PR https://github.com/ethereum/go-ethereum/pull/19207 with fix

**Comment by gballet** [comment]
#19207 has been merged, closing this issue.

**Comment by hkalodner** [comment]
I've run into this same issue myself and it appears that the reproduction example that @vsbogd provided still fails as of 1.9.4. I tested from the official docker image (`ethereum/client-go:alltools-stable`) to ensure that it wasn't a weird system config issue.

@gbalint, I'm happy to open a new issue, but I thought I'd try bumping this one since it seems like the exact same problem.

**Comment by gballet** [comment]
@hkalodner 1.9.4 has been released 11 minutes ago, so the version you were using yesterday is definitely not 1.9.4. Could you please try again with `ethereum/client-go:latest` ?

**Comment by gballet** [comment]
Oh and if you open an issue, please join your contract.json as well. Thanks.

**Comment by hkalodner** [comment]
Whoops my bad on the version. I must have transcribed it wrong. I just retried with `ethereum/client-go:latest` with abigen version `abigen version 1.9.5-unstable-24ef8351-20190919` and I still hit `Fatal: Failed to generate ABI binding: json: cannot unmarshal object into Go value of type []struct { Type string; Name string; Constant bool; Anonymous bool; Inputs []abi.Argument; Outputs []abi.Argument }`.

For the contract I just used the reproduction step in the original issue piping directly from an echo
`echo '{"contracts":{"test.sol:Test":{"abi":"[]","bin":"6080604052348015600f57600080fd5b50603580601d6000396000f3fe6080604052600080fdfea165627a7a723058204607ea1e116ed157fa353615ead6dab3b941f6c06adf7c0e724273ad8ca0a9c80029","devdoc":"{\"methods\":{}}","userdoc":"{\"methods\":{}}"}},"version":"0.5.3+commit.10d17f24.Linux.g++"}' | abigen --abi - --pkg test`
